src/docs/sphinx/dev_guide/gpu_porting.rst

```diff
@@ -0,0 +1,294 @@
+.. ## Copyright (c) 2017-2022, Lawrence Livermore National Security,con LLC and
+.. ## other Axom Project Developers. See the top-level LICENSE file for details.
+.. ##
+.. ## SPDX-License-Identifier: (BSD-3-Clause)
+
+.. _gpu-porting-label:
+
+*******************************
+GPU Porting in Axom
+*******************************
+
+Axom uses RAJA and Umpire as the main workhorses for GPU porting,
+with a set of convenience macros, Axom name-spaced wrappers around commonly
+used functions, and preset execution spaces for host/device execution.
+
+For the user's guide on using GPU utilities, see also
+:ref:`Core Acceleration<core-acceleration>`
+
+
+.. _gpu-macros-label:
+
+===============
+Macros 
+===============
+
+Link to Axom's `Macros header file`_
+
+Most of the GPU-related macros are used to guard device code before compilation
+time or for ``__host__ __device__`` decoration of functions/lambdas.
+
+For guarding device code:
+
+* ``AXOM_USE_GPU`` - Defined if Axom is built with CUDA or HIP
+
+	* ``AXOM_USE_CUDA`` - Defined if Axom is built with CUDA
+	* ``AXOM_USE_HIP`` - Defined if Axom is built with HIP
+
+* ``AXOM_DEVICE_CODE`` - Denotes code will be compiled on device
+* ``AXOM_GPUCC`` - Denotes code will be compiled with HIP or CUDA compiler
+
+For ``__host__ __device__`` decoration::
+
+  //---------------------------------------------------------------------------
+  // Functions
+  //---------------------------------------------------------------------------
+
+  #define AXOM_DEVICE __device__
+
+  #define AXOM_HOST_DEVICE __host__ __device__
+
+  #define AXOM_HOST __host__
+
+  //---------------------------------------------------------------------------
+  // Lambdas
+  //---------------------------------------------------------------------------
+
+  #define AXOM_LAMBDA [=] AXOM_HOST_DEVICE
+
+  #define AXOM_DEVICE_LAMBDA [=] AXOM_DEVICE
+
+  #define AXOM_HOST_LAMBDA [=] AXOM_HOST
+
+
+.. _gpu-memory-label:
+
+===============
+Memory 
+===============
+
+Link to Axom's `memory management header file`_
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Memory Management Routines
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Umpire has the concept of "allocators" associated with each
+`memory resource type`_ (``umpire::resource::MemoryResourceType``)
+
+To allocate memory on a particular resource, you use the ID associated with the
```

Comment:
Suggest rephrasing to make reading less awkward: "you use the ID for the allocator associated with the MemoryResourceType."